homeassistant/components/light/__init__.py

```diff
@@ -141,6 +146,7 @@ def preprocess_turn_on_alternatives(params):
     if profile is not None:
         params.setdefault(ATTR_XY_COLOR, profile[:2])
         params.setdefault(ATTR_BRIGHTNESS, profile[2])
+        params.setdefault(ATTR_TRANSITION, profile[3] if len(profile) > 3 else 0)
```

Comment:
We shouldn't set any transition if there is none. 
```suggestion
        if len(profile) > 3:
            params.setdefault(ATTR_TRANSITION, profile[3])
```